### PR TITLE
add consul client to aget

### PR DESCRIPTION
**Pull request by cgbaker**
configuration, parsing and creation of a consul client for use by the autoscaler for... whatever we decide to use it for 🤷

### CODE REVIEW

File: agent/agent.go [hunk 1]
```diff
@@ -165,6 +172,29 @@ func (a *Agent) generateNomadClient() error {
...
+	client, err := consulapi.NewClient(cfg)
+	if err != nil {
+		return fmt.Errorf("failed to instantiate Consul client: %v", err)
+	}
+	a.consulClient = client
+
+	return nil
+}
+
+
```
Comment (jrasell) [inline agent/agent.go:197]:
```suggestion
```

File: agent/config/config.go [hunk 2]
```diff
@@ -149,6 +157,169 @@ type Nomad struct {
...
+// Consul contains the configuration information necessary to
+// communicate with a Consul server.
+type Consul struct {
+	// Addr is the HTTP endpoint address of the local Consul agent
+	//
+	// Uses Consul's default and env var.
+	Addr string `hcl:"address"`
+
+	// TimeoutHCL is used by Consul HTTP Client
+	TimeoutHCL string        `hcl:"timeout" json:"-"`
```
Comment (jrasell) [inline agent/config/config.go:169]:
```suggestion
	TimeoutHCL string `hcl:"timeout" json:"-"`
```

File: agent/config/config.go [hunk 3]
```diff
@@ -149,6 +157,169 @@ type Nomad struct {
...
+
+func (c *Consul) MergeWithDefault() (*consulapi.Config, error) {
+	if c == nil {
+		return nil, nil
+	}
+
+	cfg := consulapi.DefaultConfig()
+
+	if c.TimeoutHCL != "" {
+	}
```
Comment (jrasell) [inline agent/config/config.go:217]: Does this need to be removed? I notice the same conditional is below (L225) and parses the duration.

**Review by jrasell — CHANGES_REQUESTED** [review]
Looking good.

Couple of lint failures and a question about a conditional which I think can be removed. Currently imports in the repo are only split by stdlib / non-stdlib and then sorted within these two sections.

File: agent/config/config.go (hunk 3 above)
Comment (cgbaker) [reply agent/config/config.go:217]: oops, leftover copy-pasta 🤦

**Review by lgfa29 — APPROVED** [review]
LGTM.

We tend to add CLI flags to all of our configs (like [Nomad](https://github.com/hashicorp/nomad-autoscaler/blob/main/command/agent.go#L391-L402)), but this can come in a future PR.

**Review by jrasell — APPROVED** [review]
LGTM

**Pull request merged**